### PR TITLE
feat: introduce toolchain helper

### PR DESCRIPTION
**Pull request by thesayyn**
This is not final yet, I just want to test out with rules_python first. Welcome to review though.

Also due to https://github.com/bazelbuild/bazel/issues/20297 i can't write unit tests for the incompatible flag. I'll look for a workaround.

### CODE REVIEW

**Review by comius — CHANGES_REQUESTED** [review]
> i can't write unit tests for the incompatible flag. I'll look for a workaround.

I don't mind if some tests are omitted at this level of granularity. Also the code is copied from builtin proto_common.bzl, and already tested there on a slightly bigger granularity (i.e there are tests for proto_library/proto_lang_toolchain/java_proto_library)

**Comment by thesayyn** [comment]
> I don't mind if some tests are omitted at this level of granularity. Also the code is copied from builtin proto_common.bzl, and already tested there on a slightly bigger granularity (i.e there are tests for proto_library/proto_lang_toolchain/java_proto_library)

hmm, I agree but I still believe we should test this repo with the incompatible flag to be safe even if it's not this granular.

File: proto/proto_common.bzl [hunk 1]
```diff
@@ -22,3 +22,33 @@ load("//proto/private:native.bzl", "native_proto_common")
 proto_common = native_proto_common
 
 ProtoLangToolchainInfo = proto_common.ProtoLangToolchainInfo
+
+def _incompatible_toolchains_enabled():
+    getattr(proto_common, "INCOMPATIBLE_ENABLE_PROTO_TOOLCHAIN_RESOLUTION", False)
```
[the remark this answers is not in the document]
Comment (thesayyn) [reply proto/proto_common.bzl]: Done

**Comment by thesayyn** [comment]
@comius I propose adding a new test target to bazelci here that tests the incompatible flag against a prerelease of bazel.

**Comment by thesayyn** [comment]
@comius PTAL. this is ready.

**Pull request merged**